### PR TITLE
Describe VectorMap - Custom Annotations

### PR DESCRIPTION
**Pull request by AlexJameson**
http://jsserver:81/Demos-20.2/WidgetsGallery/Demo/VectorMap/CustomAnnotations/jQuery/Light/

### CODE REVIEW

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Annotations are images and text blocks that provide additional information about the visualized data.
+
+To configure individual annotations, assign an array of objects to the [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) option. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
+
+Each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#type) attribute should be set to *"text"*, *"image"*, or *"custom"*. In this demo, the type of annotations is *"custom"*.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]:
```suggestion
Each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#type) attribute should be set to *"text"*, *"image"*, or *"custom"*. In this demo, the type of all annotations is *"custom"*.
```

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
 Annotations are images and text blocks that provide additional information about the visualized data.
```
Comment (Abadzhev) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]: Let's replace "provide" with "display". Ray mentioned several times that we overuse "provide" and not always use it appropriately. 

```suggestion
Annotations are images and text blocks that display additional information about the visualized data.
```

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+To configure annotations, assign an array of objects to the [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/) option. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
 
-Each annotation's [type](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#type) attribute should be set to *"text"* or *"image"*. In this demo, all annotations have the *"text"* type.
+Each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#type) attribute should be set to *"text"* or *"image"*. In this demo, all annotations have the *"text"* type.
```
Comment (Abadzhev) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]: The word "should" leaves some uncertainty. I "should" implies that maybe I don't do it. What happens then? If there is no way this property can be set to anything other than "text" or "image", I wouldn't use the word "should". 

 ```suggestion
You can set each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#type) attribute to *"text"* or *"image"*. This demo shows only *"text"* annotations.
```

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
 Annotations can be added at specific coordinates or anchored to a **PolarChart** element. This demo illustrates annotation placement:
```
Comment (Abadzhev) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]: Made it more active and clarified the second sentence a bit: 

```suggestion
You can place annotations at specific coordinates or anchor them to **PolarChart** elements. This demo illustrates the following annotation placement methods:  
```

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
-- [Angle](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#angle) and [radius](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#radius) options specify the position of annotations with seasons names.
-- Annotations with temperatures are anchored to series, and the [offsetX](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#offsetX), [paddingTopBottom](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#paddingTopBottom), and [paddingLeftRight](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#paddingLeftRight) options specify their position.
+- [Angle](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#angle) and [radius](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#radius) options specify the position of annotations with seasons names.
```
Comment (Abadzhev) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]: If you mean "names of seasons", then it should be "season names", i.e. first word singular. It acts like an adjective if used this way. 

```suggestion
- [Angle](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#angle) and [radius](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#radius) options specify the position of annotations with season names.
```

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
+- Annotations with temperatures are anchored to series, and the [offsetX](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#offsetX), [paddingTopBottom](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#paddingTopBottom), and [paddingLeftRight](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#paddingLeftRight) options specify their position.
 
-For more information on annotation settings, refer to the [annotations[]](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/) documentation section.
\ No newline at end of file
+For more information on annotation settings, refer to the [annotations[]](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/) documentation section.
```
Comment (Abadzhev) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]:
```suggestion
For more information on annotation settings, refer to the [annotations[]](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/) help topic.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+Annotations are images and text blocks that provide additional information about the visualized data.
```
Comment (Abadzhev) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]:
```suggestion
Annotations are images and text blocks that display additional information about the visualized data.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+To configure individual annotations, assign an array of objects to the [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) option. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
```
Comment (Abadzhev) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]: The first sentence first talks about "individual" annotations, then switches to an array (so, multiple objects). That's a bit confusing... 

```suggestion
To add annotations to a chart, assign an array of objects to the [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) option. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+Each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#type) attribute should be set to *"text"*, *"image"*, or *"custom"*. In this demo, the type of all annotations is *"custom"*.
```
Comment (Abadzhev) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]: Same comment as above - I don't like the word "should". It implies that I'm given an option to not set it at all. But then what happens?

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+
+To create an annotation of this type, declare the SVG markup in the [template](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#template) option. Inside the template, you can access the annotation being customized.
+
+For more information on annotation settings, refer to the [annotations[]](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) documentation section.
```
Comment (Abadzhev) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]:
```suggestion
For more information on annotation settings, refer to the [annotations[]](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) help topic.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
+Annotations are images and text blocks that provide additional information about the visualized data.
+
+To configure individual annotations, assign an array of objects to the [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) option. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
+
+Each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#type) attribute should be set to *"text"*, *"image"*, or *"custom"*. In this demo, the type of all annotations is *"custom"*.
+
+To create an annotation of this type, declare the SVG markup in the [template](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#template) option. Inside the template, you can access the annotation being customized.
```
Comment (Abadzhev) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]: To create an annotation of that type, you set its type to "custom" :-) In this case we already talked about type, so now we need to talk about specifying content for these custom annotations.  

```suggestion
Custom annotations require that you specify your own display [template](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#template) in SVG format. As you can see in demo code, you can access annotation data within the template markup.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md (hunk 8 above)
Comment (AlexJameson) [reply JSDemos/Demos/VectorMap/CustomAnnotations/description.md]: I applied your suggestions, but let me clarify what I meant.

Each object in the annotations[] array represents an individual annotations, so users have to specif individual settings in these objects.

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 12]
```diff
@@ -1,12 +1,12 @@
-Annotations are images and text blocks that provide additional information about the visualized data.
+Annotations are containers for images, text blocks and custom content that display additional information about the visualized data.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]:
```suggestion
Annotations are containers for images, text blocks, and custom content that display additional information about the visualized data.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 13]
```diff
@@ -0,0 +1,9 @@
+Annotations are containers for images, text blocks and custom content that display additional information about the visualized data.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]:
```suggestion
Annotations are containers for images, text blocks, and custom content that display additional information about the visualized data.
```

File: JSDemos/Demos/Charts/PolarChartAnnotations/description.md [hunk 14]
```diff
@@ -1,12 +1,12 @@
...
+To create annotations, populate the widget's [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/) array. Each object in the array configures an individual annotation. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
 
-Each annotation's [type](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#type) attribute should be set to *"text"* or *"image"*. In this demo, all annotations have the *"text"* type.
+You can set each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#type) attribute to *"text"*, *"image"*, or *"custom"*. This demo shows only *"text"* annotations.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Charts/PolarChartAnnotations/description.md]:
```suggestion
You can set each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/#type) option to *"text"*, *"image"*, or *"custom"*. This demo shows only *"text"* annotations.
```

File: JSDemos/Demos/VectorMap/CustomAnnotations/description.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
...
+
+To create annotations, populate the widget's [annotations](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/) array. Each object in the array configures an individual annotation. To specify settings for all annotations, use the [commonAnnotationSettings](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
+
+You can set each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#type) attribute to *"text"*, *"image"*, or *"custom"*. In this demo, the type of all annotations is *"custom"*.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/VectorMap/CustomAnnotations/description.md]:
```suggestion
You can set each annotation's [type](/Documentation/ApiReference/Data_Visualization_Widgets/dxVectorMap/Configuration/annotations/#type) option to *"text"*, *"image"*, or *"custom"*. In this demo, the type of all annotations is *"custom"*.
```

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**